src/azure-cli-core/azure/cli/core/test_utils/vcr_test_base.py

```diff
@@ -412,39 +420,46 @@ def execute(self):
+    def __init__(self, test_file, test_name, resource_group='vcr_resource_group', run_live=False,
+                 debug=False, debug_vcr=False, skip_setup=False, skip_teardown=False):
         super(ResourceGroupVCRTestBase, self).__init__(test_file, test_name, run_live=run_live,
-                                                       debug=debug, skip_setup=skip_setup,
+                                                       debug=debug, debug_vcr=debug_vcr,
+                                                       skip_setup=skip_setup,
                                                        skip_teardown=skip_teardown)
-        self.resource_group = 'vcr_resource_group'
+        self.resource_group_original = resource_group
+        random_tag = '_{}_'.format(''.join((choice(ascii_lowercase + digits) for _ in range(4))))
+        self.resource_group = '{}{}'.format(resource_group, '' if self.playback else random_tag)
         self.location = 'westus'
 
     def set_up(self):
-        self.cmd('resource group create --location {} --name {}'.format(
+        self.cmd('resource group create --location {} --name {} --tags use=az-test'.format(
             self.location, self.resource_group))
 
     def tear_down(self):
-        self.cmd('resource group delete --name {}'.format(self.resource_group))
+        self.cmd('resource group delete --name {} --no-wait'.format(self.resource_group))
 
 class StorageAccountVCRTestBase(VCRTestBase):
     # pylint: disable=too-many-arguments
-    def __init__(self, test_file, test_name, run_live=False, debug=False, debug_vcr=False,
-                 skip_setup=False, skip_teardown=False):
+    def __init__(self, test_file, test_name, resource_group='vcr_resource_group', run_live=False,
+                 debug=False, debug_vcr=False, skip_setup=False, skip_teardown=False):
         super(StorageAccountVCRTestBase, self).__init__(test_file, test_name, run_live=run_live,
-                                                        debug=debug, skip_setup=skip_setup,
+                                                        debug=debug, debug_vcr=debug_vcr,
+                                                        skip_setup=skip_setup,
                                                         skip_teardown=skip_teardown)
-        self.resource_group = 'vcr_resource_group'
+        self.resource_group_original = resource_group
+        random_tag = '_{}_'.format(''.join((choice(ascii_lowercase + digits) for _ in range(4))))
+        self.resource_group = '{}{}'.format(resource_group, '' if self.playback else random_tag)
+
         self.account = MOCKED_STORAGE_ACCOUNT if self.playback else \
             'vcrstorage{}'.format(''.join(choice(digits) for i in range(12)))
         self.location = 'westus'
 
     def set_up(self):
-        self.cmd('resource group create --location {} --name {}'.format(
+        self.cmd('resource group create --location {} --name {} --tags use=az-test'.format(
```

Comment:
_This is a quick thought, i don't intend to trigger any additional change or delay the PR here._

Could we call the API of the SDK directly here since this is just a set up step?